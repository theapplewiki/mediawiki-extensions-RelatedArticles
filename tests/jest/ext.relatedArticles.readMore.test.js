const { render } = require( '../../resources/ext.relatedArticles.readMore/index.js' );

describe( 'ext.relatedArticles.readMore.bootstrap', () => {
	it( 'renders without error', () => {
		const element = document.createElement( 'div' );

		render( [
			{
				title: 'Hello no description'
			},
			{
				title: 'Hello',
				description: 'Description',
				thumbnail: {
					source: 'hello.gif',
					width: 200,
					height: 200
				}
			},
			{
				title: 'Hello with extract',
				extract: 'Extract',
				thumbnail: {
					source: 'helloEx.gif',
					width: 200,
					height: 200
				}
			},
			{
				title: 'Hello with pageprops',
				pageprops: {
					description: 'Page props desc'
				}
			}
		], element );
		expect( element.innerHTML ).toMatchSnapshot();
	} );
} );
