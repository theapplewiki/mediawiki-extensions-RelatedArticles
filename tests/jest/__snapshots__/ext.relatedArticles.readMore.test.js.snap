// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ext.relatedArticles.readMore.bootstrap renders without error 1`] = `"<aside class=\\"ra-read-more noprint\\"><h2>relatedarticles-read-more-heading</h2><ul class=\\"ext-related-articles-card-list\\"><li title=\\"Hello no description\\" class=\\"ext-related-articles-card\\"><div class=\\"ext-related-articles-card-thumb ext-related-articles-card-thumb-placeholder\\"></div><a href=\\"Hello no description\\" aria-hidden=\\"true\\" tabindex=\\"-1\\"></a><div class=\\"ext-related-articles-card-detail\\"><h3><a href=\\"Hello no description\\">Hello no description</a></h3><p class=\\"ext-related-articles-card-extract\\"></p></div></li><li title=\\"Hello\\" class=\\"ext-related-articles-card\\"><div class=\\"ext-related-articles-card-thumb\\" style=\\"background-image: url(hello.gif);\\"></div><a href=\\"Hello\\" aria-hidden=\\"true\\" tabindex=\\"-1\\"></a><div class=\\"ext-related-articles-card-detail\\"><h3><a href=\\"Hello\\">Hello</a></h3><p class=\\"ext-related-articles-card-extract\\">Description</p></div></li><li title=\\"Hello with extract\\" class=\\"ext-related-articles-card\\"><div class=\\"ext-related-articles-card-thumb\\" style=\\"background-image: url(helloEx.gif);\\"></div><a href=\\"Hello with extract\\" aria-hidden=\\"true\\" tabindex=\\"-1\\"></a><div class=\\"ext-related-articles-card-detail\\"><h3><a href=\\"Hello with extract\\">Hello with extract</a></h3><p class=\\"ext-related-articles-card-extract\\">Extract</p></div></li><li title=\\"Hello with pageprops\\" class=\\"ext-related-articles-card\\"><div class=\\"ext-related-articles-card-thumb ext-related-articles-card-thumb-placeholder\\"></div><a href=\\"Hello with pageprops\\" aria-hidden=\\"true\\" tabindex=\\"-1\\"></a><div class=\\"ext-related-articles-card-detail\\"><h3><a href=\\"Hello with pageprops\\">Hello with pageprops</a></h3><p class=\\"ext-related-articles-card-extract\\">Page props desc</p></div></li></ul></aside>"`;
